src/java.base/share/classes/sun/security/ssl/SSLSessionImpl.java

```diff
@@ -356,7 +356,7 @@ final class SSLSessionImpl extends ExtendedSSLSession {
         }
 
         // PSK identity
-        i = buf.get();
+        i = Byte.toUnsignedInt(buf.get());
```

Comment:
Could you rewrite this constructor to use the helper methods from `sun.security.ssl.Record` instead of reading data from the buffer directly?